webapp/src/app/components/leaderboard/leaderboard.component.html

```diff
@@ -0,0 +1,55 @@
+<div class="flex flex-col gap-4 py-8">
+  <h1 class="text-3xl font-bold">Artemis Leaderboard</h1>
```

Comment:
Maybe just have a leaderboard page component that is the smart component and separate out the table, I think this is the way. 

We can skip having a story for the smart components for now, there we would need to mock the services in the stories.

Reply:
Right now the leaderboard is already basically just a wrapper for the `table`-component with the required logic it feels like.
I guess we could also just move the heading to the `main`-component to make the separation more clear?
Is there an advantage to separate components I'm missing here?

Reply:
I'm still thinking about it 🤔 Okay maybe taking a step back this page would be called the home page or the dashboard page which fetches the data from the service(s) using a leaderboard component. But ideally the leaderboard component is dumb and does not require a dependency to a service, because this always makes testing or showing it in the storybook harder.

If we were to add for example some user actions for filtering the timeframe etc. the leaderboard component would have a callback/output/binding to do that but does not smartly also set the filter for the service. Does this make sense? 

In the design system `/ui` the components should be rather generic and reusable. Having a more specialized component building the leaderboard using the code from the design system which also has a story also makes sense to me. I would like to show the leaderboard component in the storybook for documentation but we do not necessarily have to show the whole home page / dashboard page smart component in there.

Reply:
I think I understand where you are going with that idea now. I personally didn't think about that separation right away because we don't really plan on creating multiple leaderboards rather than one for all repos.

With your explanation I do see the point for it, although I feel like it creates an (unnecessary?) level of abstraction in this use case. Let's discuss further this on Monday maybe and rework in a follow-up PR?

Reply:
Let's do it in a follow-up. In my mind the leaderboard is just one component on the page. We will add more gamification elements on the same page but you are also right that we might create a leaderboard route per repo in the near future 🤔